crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -0,0 +1,335 @@
+            } else {
+                typed_store::rocks::open_cf_opts(path, db_options, opt_cfs)
+            }
+        }
+        .expect("Cannot open DB.");
+
+        let executed_sequence =
+            DBMap::reopen(&db, Some(EXEC_SEQ_TABLE_NAME)).expect("Cannot open CF.");
+
+        let (
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            assigned_object_versions,
+            next_object_versions,
+            consensus_message_processed,
+            batches,
+            last_consensus_index,
+            epochs,
+        ) = reopen! (
+            &db,
+            OBJECTS_TABLE_NAME;<ObjectKey, Object>,
+            OWNER_INDEX_TABLE_NAME;<(Owner, ObjectID), ObjectInfo>,
+            TX_TABLE_NAME;<TransactionDigest, TransactionEnvelope<S>>,
+            CERTS_TABLE_NAME;<TransactionDigest, CertifiedTransaction>,
+            PENDING_EXECUTION;<InternalSequenceNumber, TransactionDigest>,
+            PARENT_SYNC_TABLE_NAME;<ObjectRef, TransactionDigest>,
+            EFFECTS_TABLE_NAME;<TransactionDigest, TransactionEffectsEnvelope<S>>,
+            ASSIGNED_OBJECT_VERSIONS_TABLE_NAME;<(TransactionDigest, ObjectID), SequenceNumber>,
+            NEXT_OBJECT_VERSIONS_TABLE_NAME;<ObjectID, SequenceNumber>,
+            CONSENSUS_MESSAGE_PROCESSED_TABLE_NAME;<TransactionDigest, bool>,
+            BATCHES_TABLE_NAME;<TxSequenceNumber, SignedBatch>,
+            LAST_CONSENSUS_TABLE_NAME;<u64, ExecutionIndices>,
+            EPOCH_TABLE_NAME;<EpochId, AuthenticatedEpoch>
+        );
+
+        Self {
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            assigned_object_versions,
+            next_object_versions,
+            consensus_message_processed,
+            executed_sequence,
+            batches,
+            last_consensus_index,
+            epochs,
+        }
+    }
+
+    /// Open an authority store by directory path in read-write mode
+    pub fn open_read_write<P: AsRef<Path>>(path: P, db_options: Option<Options>) -> Self {
+        Self::open_impl(path, db_options, None)
+    }
+
+    /// Open an authority store by directory path in read only mode
+    pub fn open_read_only<P: AsRef<Path>>(
+        path: P,
+        secondary_path: P,
+        db_options: Option<Options>,
+    ) -> Self {
+        Self::open_impl(path, db_options, Some(secondary_path))
+    }
+
+    // TODO: condense with macros
+    pub fn dump(&self, table_name: &str) -> anyhow::Result<BTreeMap<String, String>> {
+        Ok(match table_name {
+            OBJECTS_TABLE_NAME => {
+                self.objects.try_catch_up_with_primary()?;
+                self.objects
+                    .iter()
```

Comment:
We need to put a limit here.  This will work today, but when we have millions of TXes, that BTreeMap you're building will quickly run out of memory.   I'd rather we quickly change the limits than build something which does not have limits and may OOM.

Also, it wouldn't be much work to put in a trait like `DumpableDBMap` in `typed_store` crate which implements a Dump function for each key and value.  Then, we would not need the match, and any table that implements `DumpableDBMap` could easily be dumped.   If the API is good, we also would not need to dump everything out.

Reply:
> We need to put a limit here. This will work today, but when we have millions of TXes, that BTreeMap you're building will quickly run out of memory. I'd rather we quickly change the limits than build something which does not have limits and may OOM.
> 
> Also, it wouldn't be much work to put in a trait like `DumpableDBMap` in `typed_store` crate which implements a Dump function for each key and value. Then, we would not need the match, and any table that implements `DumpableDBMap` could easily be dumped. If the API is good, we also would not need to dump everything out.

For the trait, yes that's the plan. This will be introduced as we add support to more 

> Ok, so I realize now that this change is basically
> 
> 1. Porting the older db_dump code into a store_tables
> 2. and separating out store_tables from `authority_store`.
> 
> I have two thoughts:
> 
> * Sooner or later, I'm going to refactor most of these tables into separate stores like an ObjectStore, CertificateStore etc.   So, while this move of tables to a separate StoreTables is functionally fine, I wonder if it would be better to minimize the changes now and keep dump() where it is without moving all the tables, as stuff will get refactored later anyways
> * See my other comment but it wouldn't be much more work to create a trait like `DumpableDBMap` which would make the code much cleaner without having to match on every table.   I think it could be doable without macros....
> 
> Anyways my request for change mainly has to do with my other comment about limits.

The reason for this code change is to reduce code duplication 
I agree we should factor out the stores. But for now we can go incrementally.

Regarding using a trait, yes that was the initial plan and it's a good plobut it still puts dump logic in Sui. I decided this is a feature we will want in narhwal too so best if its part of typedstore natively

Reply:
For limits, you want us to limit how much data we dump?

Reply:
It's just we want to avoid having data structures that can grow unbounded as a principle.
Just `.iter().take(LIMIT)` should suffice for now.

We can discuss pagination later, agreed that should be pushed off.

Reply:
This PR will be used to track the size limit improvement 
https://github.com/MystenLabs/sui/issues/3532